src/litdata/utilities/dataset_utilities.py

```diff
@@ -115,3 +114,57 @@ def generate_roi(chunks: List[Dict[str, Any]], item_loader: Optional[BaseItemLoa
             roi.append((0, end))
 
     return roi
+
+
+def load_index_file(input_dir: str) -> Dict[str, Any]:
+    """Load index file from the input_dir."""
+
+    index_filepath = os.path.join(input_dir, _INDEX_FILENAME)
+    try:
+        # load index.json file
+        with open(index_filepath) as f:
+            data = json.load(f)
+            if "chunks" not in data:
+                raise KeyError(f"'chunks' not found in the index file at {index_filepath}.")
+            return data
+    except KeyError as e:
```

Comment:
Instead of relying on try-except, the better approach will be to `use conditionals (if, elif, else)`.

If tomorrow we have to add support for another StreamingDataset, we will be required to make a nested try-except code which will be difficult to read.

Also, extract out the logic of reading mds index_json files into another function. Let's keep this function clean.

```python
# something like

if "chunks" in data:
  # litdata optimize
  return data
elif "shards" in data:
  # mds writer
  return read_mds_index_json()
else:
  ...
```